sparkmagic/sparkmagic/tests/test_configuration.py

```diff
@@ -12,19 +13,19 @@ def _setup():
 
 @with_setup(_setup)
 def test_configuration_override_base64_password():
-    kpc = { 'username': 'U', 'password': 'P', 'base64_password': 'cGFzc3dvcmQ=', 'url': 'L' }
+    kpc = { 'username': 'U', 'password': 'P', 'base64_password': 'cGFzc3dvcmQ=', 'url': 'L', "auth_type": AUTH_BASIC }
     overrides = { conf.kernel_python_credentials.__name__: kpc }
     conf.override_all(overrides)
     conf.override(conf.status_sleep_seconds.__name__, 1)
     assert_equals(conf.d, { conf.kernel_python_credentials.__name__: kpc,
                                      conf.status_sleep_seconds.__name__: 1 })
     assert_equals(conf.status_sleep_seconds(), 1)
-    assert_equals(conf.base64_kernel_python_credentials(), { 'username': 'U', 'password': 'password', 'url': 'L' })
+    assert_equals(conf.base64_kernel_python_credentials(), { 'username': 'U', 'password': 'password', 'url': 'L', 'auth_type': AUTH_BASIC })
 
 
 @with_setup(_setup)
 def test_configuration_override_fallback_to_password():
-    kpc = { 'username': 'U', 'password': 'P', 'url': 'L' }
+    kpc = { 'username': 'U', 'password': 'P', 'url': 'L', 'auth_type': None }
```

Comment:
We need some extra tests:

1. when auth_type is not there at all in the config file, `conf.base64_kernel_python_credentials` (and all other languages) should return `auth_type` as `constants.NO_AUTH` if the username and password are the empty string
1. when auth_type is not there at all in the config file, `conf.base64_kernel_python_credentials` (and all other languages) should return `auth_type` as `constants.AUTH_BASIC` if the username or password are not the empty string

This will ensure backwards compatibility to existing `config.json` files.